airflow/providers/amazon/aws/operators/rds.py

```diff
@@ -0,0 +1,605 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import json
+import time
+from typing import TYPE_CHECKING, List, Optional, Sequence
+
+from mypy_boto3_rds.type_defs import TagTypeDef
+
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.providers.amazon.aws.hooks.rds import RdsHook
+from airflow.providers.amazon.aws.utils.rds import RdsDbType
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+class BaseRdsOperator(BaseOperator):
+    """Base operator that implements common functions for all operators"""
+
+    ui_color = "#eeaa88"
+    ui_fgcolor = "#ffffff"
+
+    def __init__(self, *args, aws_conn_id: str = "aws_conn_id", hook_params: Optional[dict] = None, **kwargs):
+        hook_params = hook_params or {}
+        self.hook = RdsHook(aws_conn_id=aws_conn_id, **hook_params)
+        super().__init__(*args, **kwargs)
+
+        self._wait_interval = 60  # seconds
+
+    def _describe_item(self, **kwargs) -> list:
+        """Returns information about target item: snapshot, task or event"""
+        raise NotImplementedError
+
+    def _await_termination(
```

Comment:
seems this is more generic than waiting just for termination -- could it also be used for waiting for cluster to come online?  if so perhaps a more generic name is appropriate e.g. `_await_status` or something?

Reply:
Very good point.
I chose this name accidentally because I saw something similar in the Google cloud provider.
Renamed